### DISCUSSION TITLE
Rust analyzer fails to do certain type inference

### DISCUSSION THREAD

**Discussion by debasish-raychawdhuri** [rust-lang/rust-analyzer#14953] [General]
I am using rust-analyzer with both vscode and neovim. I have observed sometimes rust-analyzer fails to do type inference (and hence loses further auto-completion) for certain types. For example - the following code 

```rust
fn extract_text_field(body: &str, regex: &str) -> Option<String> {
    let re = Regex::new(regex).unwrap();
    let text = re.find(body)?.as_str();
    Some(text.to_string())
}
```
The ```Regex::new``` has the correct type resolved, but after doing unwrap, the type of ```re``` is unknown as per rust-analyzer. However, the compiler works just fine.

**Comment by flodiebold** [comment]
Yeah, that shouldn't happen. Can you reproduce this in a clean project with no dependencies other than regex? (I assume `Regex` is from the `regex` crate?)

**Comment by debasish-raychawdhuri** [comment]
Yes Regex is from regex crate. It shows the same behaviour with this project that I just created with only regex as dependency.
https://github.com/debasish-raychawdhuri/ra_test

**Comment by flodiebold** [comment]
I can't reproduce that. What Rust version are you using?

**Comment by debasish-raychawdhuri** [comment]
Rust version ```rustc 1.69.0 (84c898d65 2023-04-16)```
 ```rust-analyzer 1.69.0 (84c898d 2023-04-16)```

**Comment by debasish-raychawdhuri** [comment]
Problem got fixed after updating everything to 1.70. Thanks.
